spark-3.1-spanner-lib/src/main/java/com/google/cloud/spark/spanner/SpannerUtils.java

```diff
@@ -192,6 +194,18 @@ public static InternalRow spannerStructToInternalRow(
     return sparkRow;
   }
 
+  public String serializeMap(Map<String, String> m) throws Exception {
+    ObjectMapper mapper = new ObjectMapper();
+    return mapper.writerWithDefaultPrettyPrinter().writeValueAsString(m);
```

Comment:
PrettyPrinter adds makes the messages larger which may impact performance. Is it necessary?

Reply:
Removed, thanks David!